planner_server:
  ros__parameters:
    expected_planner_frequency: 60.0
    costmap_update_timeout: 0.2
    service_introspection_mode: "disabled"
    planner_plugins: ['GridBased']
    GridBased:
      plugin: 'nav2_smac_planner/SmacPlannerLattice'
      tolerance: 0.1
      use_sim_time: false
      downsample_costmap: false
      downsample_factor: 1
      max_planning_time : 0.05
      allow_unknown: true
      max_on_approach_iterations: 10
      allow_unknown: false
      rotation_penalty: 0.0


global_costmap/global_costmap:
  ros__parameters: 
    update_frequency: 20.0
    publish_frequency: 20.0
    footprint: "[[0.45, 0.32],[0.45,-0.32],[-0.45,-0.32],[-0.45,0.32]]"  # Robot footprint
    global_frame: base_footprint
    robot_base_frame: base_footprint
    use_sim_time: False
    rolling_window: True  # Enable rolling window for the local costmap
    width: 6  # Size of the local costmap (around the robot)
    height: 6
    resolution: 0.06
    plugins: ["obstacle_layer", "inflation_layer"]  # Add desired layers
    obstacle_layer:
      plugin: "nav2_costmap_2d::ObstacleLayer"
      enabled: true
      footprint_clearing_enabled: false
      observation_sources: scan  # Subscribe to scan data
      footprint_clearing_enabled: true
      max_obstacle_height: 2.0
      combination_method: 0
      scan:
        topic: /scan
        sensor_frame: base_footprint
        obstacle_max_range: 3.0
        obstacle_min_range: 0.0
        raytrace_max_range: 5.0
        raytrace_min_range: 0.0
        max_obstacle_height: 2.0
        min_obstacle_height: 0.0
        clearing: true
        marking: true
        data_type: "LaserScan"
    inflation_layer:
      plugin: "nav2_costmap_2d::InflationLayer"
      enabled: true
      inflation_radius: 0.3
      cost_scaling_factor: 1.0
      inflate_unknown: false
      inflate_around_unknown: true
